CRM/Contribute/Form/ManagePremiums.php

```diff
@@ -287,15 +287,16 @@ public function postProcess() {
       $params[$field] = CRM_Utils_Rule::cleanMoney($params[$field]);
     }
 
-    $ids = array();
+    $productId = NULL;
+    // If we're updating, we need to pass in the product Id
     if ($this->_action & CRM_Core_Action::UPDATE) {
-      $ids['premium'] = $this->_id;
+      $productId = $this->_id;
     }
 
     $this->_processImages($params);
 
-    // Save to database
-    $premium = CRM_Contribute_BAO_ManagePremiums::add($params, $ids);
```

Comment:
This seems like a change in the signature.

Reply:
It is. Very similar to a change @eileenmcnaughton made elsewhere recently.  This is the only place that function is called in core.

Reply:
@mattwire for standard 'add' functions we should try to drop the second param altogether (the one I changed was an oddball not a standard one - seems like here we could pass $params['id'] in this case though - isn't $productId a pseudonym for premiumId?

Reply:
> seems like here we could pass $params['id'] in this case though - isn't $productId a pseudonym for premiumId?

That was what I thought but it's not.  The premiumId is the link to the contribution page (in civicrm_premiums) while the productId is the id of the premium product in civicrm_product.
I thought about adding it to the $params object but that currently represents the "premium" and not the "product" and I didn't want to mix the two - what do you think @eileenmcnaughton ?

Reply:
@mattwire how about product_id in the params array then?

Reply:
sorry - the reverse ['premium_id']

FWIW I would support creating a NEW BAO object for product & looking at some deprecation - since this if very ambiguous 

class CRM_Contribute_BAO_ManagePremiums extends CRM_Contribute_DAO_Product

Reply:
Right.  So I've cleaned this up a little further (adding a new BAO is taking the scope too far for now @eileenmcnaughton but I agree :-) ).
- I've removed the second parameter to `CRM_Contribute_BAO_ManagePremiums::add` and replaced it with `$productParams['id']`.
- To make it easier to understand in future I've also renamed the `$params` array to `$productParams` wherever it makes sense - as this part of the code confused both myself and @eileenmcnaughton